src/nupic/research/spatial_pooler.py

```diff
@@ -19,13 +19,12 @@
 # http://numenta.org/licenses/
 # ----------------------------------------------------------------------
 
-import itertools
-
 import numpy
 from nupic.bindings.math import (SM32 as SparseMatrix,
                                  SM_01_32_32 as SparseBinaryMatrix,
                                  GetNTAReal,
                                  Random as NupicRandom)
+import nupic.math.topology as topology
```

Comment:
I don't know how this accomplishes that but, regardless, we don't use this style of import so please change.